sys/net/application_layer/dns/dns.c

```diff
@@ -168,11 +168,9 @@ static int _parse_dns_reply(uint8_t *buf, size_t len, void* addr_out, int family
     return -1;
 }
 
+static uint8_t dns_buf[SOCK_DNS_BUF_LEN];
```

Comment:
*If* you want to make this global you need to protect it now. If two threads call `sock_dns_query()` now, they are in a race for the buffer space. I would prefer keeping it in the stack though (less complicated and doesn't lead to side effects).

Reply:
I moved it back on the stack.